source/elements/oneDAL/source/data_management/index.rst

```diff
@@ -126,113 +130,93 @@ used at the data preparation and data processing stages for the following:
+Table metadata
+--------------
 
-- The :capterm:`data type` of each feature (e.g. :code:`float` or
-  :code:`double`).
+Table metadata concept contains an additional information about data into the
+table, that is not included into the table itself but can be useful in the
+particular use case.
 
-.. note::
-  Metadata can contain both compile-time and run-time information. For example,
-  basic compile-time metadata is the type of a dataset - whether it is a
-  particular :txtref:`data-source` or a :txtref:`table`. Run-time information
-  can contain the :capterm:`feature` types and :capterm:`data types <data type>`
-  of a dataset.
-
-.. _table-builder:
-
-Table builder
--------------
-
-A table :capterm:`builder` is a concept that is associated with a particular
-:txtref:`table` type and is used at the data preparation and data processing
-stages for:
-
-- Iterative construction of a :txtref:`table` from another :txtref:`<table>s` or
-  a different in-memory :txtref:`dataset` representations.
+1. Table metadata includes information about :capterm:`data types <Data type>`
+of table columns.
 
-- Construction of a :txtref:`table` from different entities that hold blocks of
-  data, such as arrays, pointers to the memory, external entities.
+2. Table metadata includes information about logical types of columns:
+   :capterm:`nominal <Nominal feature>`, :capterm:`ordinal <Ordinal feature>`,
+   :capterm:`interval <Interval feature>`, or :capterm:`ratio <Ratio feature>`.
+   Usually we speak about logical types, when the table represents a dataset.
+   But this logical type information can be also applied on the other cases,
+   e.g. when the table represents a matrix.
 
-- Changing dataset values. Since :txtref:`table` is an
-  :capterm:`immutable <immutability>` dataset, a builder provides the ability to
-  change the values in a dataset under construction.
+.. warning::
+  While extending the table concept, specification implementer should
+  distinguish whether a new property he or she is adding is the property of
+  particular ``table`` sub-type or it is a property of table metadata.
 
-- Encapsulating construction process of a :txtref:`table`. This is used to hide the
-  implementation details as they are irrelevant for users. This also allow to
-  select the most appropriate table implementation for each particular case.
+  For example, :capterm:`data layout` and :capterm:`data format` are properties
+  of table objects since they affect the structure of a table, its contract and
+  behavior.
 
-- Providing additional information on how to create a :txtref:`table` inside an
-  algorithm for :txtref:`<result>s`. This information includes metadata, memory
-  allocators that shall be used, or even a particular table implementation.
-
-For details, see :txtref:`table-builders` section.
+  Logical and physical types of table columns are related strongly to the data
+  in the table and do not affect table concept definition - that's why they are
```

Comment:
it looks like we explain some rationale/difference between table&metadata. It would be better to briefly discuss it in the def section and do not cover it here